### PR TITLE
dialects: (memref) add custom syntax to memref.alloc

### PR DESCRIPTION
**Pull request by superlopuh**

### CODE REVIEW

File: xdsl/dialects/memref.py [hunk 1]
```diff
@@ -147,15 +149,29 @@ class Alloc(IRDLOperation):
...
+        result_type: Attribute,
+        alignment: Attribute | None = None,
+    ):
+        super().__init__(
+            operands=(dynamic_sizes, symbol_operands),
+            result_types=(result_type,),
+            properties={"alignment": alignment},
+        )
+
+    @classmethod
```
Comment (superlopuh) [inline xdsl/dialects/memref.py:165]: can deprecate this in the future

Comment (JosseVanDelm) [reply xdsl/dialects/memref.py:165]: Dumb question: Why is this suddenly a classmethod?

File: xdsl/dialects/memref.py [hunk 2]
```diff
@@ -185,6 +200,67 @@ def verify_(self) -> None:
                 "op dimension operand count does not equal memref dynamic dimension count."
             )
 
+    def print(self, printer: Printer):
+        printer.print_string("(")
+        printer.print_list(self.dynamic_sizes, printer.print_ssa_value)
+        printer.print_string(")")
```
Comment (JosseVanDelm) [inline xdsl/dialects/memref.py:206]: Another dumb question, is this not possible with custom-format printing/parsing?

**Review by JosseVanDelm — APPROVED** [review]
Looks good, I just have some questions

File: xdsl/dialects/memref.py (hunk 2 above)
Comment (superlopuh) [reply xdsl/dialects/memref.py:206]: I don't think we handle not printing the delimiters of an optional list if it's empty. CC @math-fehr @PapyChacal is this correct?

File: xdsl/dialects/memref.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/memref.py:165]: That was a remainder of a WIP change, I can change it back now, I just felt like it didn't hurt.

File: xdsl/dialects/memref.py (hunk 2 above)
Comment (math-fehr) [reply xdsl/dialects/memref.py:206]: Soon TM, for now we can't do that :(
But @PapyChacal is working on this AFAIK

**Pull request merged**